### PR TITLE
feat: timer field

### LINKED ISSUE #4083: [FR] Time field type
### Description

A field type for determining time like 20 minutes, 3 hours, etc. It's useful for task management purposes, To show how much time a task needs or how much I've spent time on the task.

### Impact

Users who use the Kanban or grid view for managing tasks or similar stuff.

### Additional Context

I would like to implement this my self if its an acceptable feature.

### PR DESCRIPTION
**Pull request by zoli**
Timer field type which stores time in minutes. More details about this feature on the related issue #4083.
Resolves #4083.

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by zoli** [comment]
https://github.com/AppFlowy-IO/AppFlowy/assets/3286303/19c5d423-9f5b-4d12-8940-8a3f9dd62ea2

File: frontend/rust-lib/flowy-database2/src/services/field/type_options/timer_type_option/timer.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    Ok(TimerCellData::from(cell))
+  }
+}
+
+impl TimerTypeOption {
+  pub fn new() -> Self {
+    Self
+  }
+
+  pub(crate) fn format_cell_data(minutes: i64) -> String {
```
Comment (Xazin) [inline frontend/rust-lib/flowy-database2/src/services/field/type_options/timer_type_option/timer.rs:63]: For this use-case, I'm wondering if we shouldn't do the parsing of the cell in the frontend itself.

Using Intl I believe we can do some pretty nice transformation, including support for different localizations.

See eg. https://github.com/andresaraujo/timeago.dart/blob/master/packages/timeago/lib/src/timeago.dart

**Comment by Xazin** [comment]
Also something I was thinking about when I saw "Timer", is that I was expecting it to count down. Otherwise I guess calling it "Time" is more relevant.

File: frontend/rust-lib/flowy-database2/src/services/field/type_options/timer_type_option/timer.rs (hunk 1 above)
Comment (zoli) [reply frontend/rust-lib/flowy-database2/src/services/field/type_options/timer_type_option/timer.rs:63]: I wasn't sure about this either but doesn't this make it prone to inconsistency between different frontends?

**Comment by zoli** [comment]
> Also something I was thinking about when I saw "Timer", is that I was expecting it to count down. Otherwise I guess calling it "Time" is more relevant.

You are right about timer but I think maybe stopwatch is a better name. I also want to add time tracking feature to this later and it works like count up.

**Comment by zoli** [comment]
Thinking about this, It can be a general field type (I'm not sure about the name) which the user can choose to be a timer (count down) or stopwatch (count up) in type options. What do you think @Xazin?

**Comment by Xazin** [comment]
> Thinking about this, It can be a general field type (I'm not sure about the name) which the user can choose to be a timer (count down) or stopwatch (count up) in type options. What do you think @Xazin?

Yeah I that is a good idea. Kind of like selecting a number format.

**Comment by zoli** [comment]
So I will change the field type name to Time and will add a stopwatch and timer option later in a separate PR.

File: frontend/rust-lib/flowy-database2/src/services/field/type_options/timer_type_option/timer.rs (hunk 1 above)
Comment (Xazin) [reply frontend/rust-lib/flowy-database2/src/services/field/type_options/timer_type_option/timer.rs:63]: You're right, but I think in this case it's the better solution.

In the frontend we have locale information, as well as a better Intl library so we can much easilier parse the time into eg. `5y 100d 2h 31m`. With timestamps we also do parsing on the frontend for dates, we also are able to get timezone information on the frontend quite easily.

In Danish the equivalent would be `5år 100d 2t 31m`.

**Comment by Xazin** [comment]
> So I will change the field type name to Time and will add a stopwatch and timer option later in a separate PR.

Sounds good, feel free to ping me when you have renamed the field, I tested and feature works as expected. 👍

I think the time type data can be in minutes, but for Stop timer or Count down timer, it should be a start timestamp.

**Comment by zoli** [comment]
@Xazin I renamed the field type to time and moved the formatting function to the flutter frontend.
I will implement these features on other PRs:
- Having precision option (store by minutes or second)
- Timer and Stopwatch as type option
- More localization support as per your review comment.
- Tracking time button for timer and stopwatch

**Pull request merged**